libraries/iterator.go

```diff
@@ -82,7 +89,9 @@ func (i *mergedLocationIter) Next() (borges.Location, error) {
 			return nil, err
 		}
 
+		i.mu.Lock()
 		i.iters = i.iters[1:]
+		i.mu.Unlock()
```

Comment:
Shouldn't it protect the whole function so the rest of hash accesses are in the lock? These:

```go
if len(i.iters) == 0 {

loc, err := i.iters[0].Next()
```

Reply:
Yep, you are right, I just narrowed locks to fix race, but the whole method should be locked